File: JwtAuthDotNet9/Services/TransactionService.cs
using JwtAuthDotNet9.Data;
using JwtAuthDotNet9.Entities;
using JwtAuthDotNet9.Services.IServico;
using Microsoft.EntityFrameworkCore;

namespace JwtAuthDotNet9.Services
{
    public class TransactionService(UserDbContext context) : ITransactionService
    {
        public async Task CreateAsync(Transaction transaction)
        {
            context.Transaction.Add(transaction);
            await context.SaveChangesAsync();
        }

        public async Task<List<Transaction>> ListAllTransaction(Guid WalletID)
        {
            Task<List<Transaction>>? transactions = context.Transaction.Where(t => t.WalletInformationId == WalletID).ToListAsync();

            return await transactions;
        }

        public async Task<List<Transaction>> ListAllTagTransaction(Guid WalletID)
        {
            List<Transaction>? taggedTransactions = await context.Transaction.Where(t => t.WalletInformationId == WalletID && !string.IsNullOrEmpty(t.TagName)).ToListAsync();

            return taggedTransactions;
        }
    }
}

File: JwtAuthDotNet9/frontend/src/services/transactionService.js
// services/transactionService.js
import { api } from './api'

export async function getTags(walletId) {
    const response = await api.post('/api/transaction/listalltag', {
        WalletInformationId: walletId
    })

    return response.data.map(tx => ({
        id: tx.id,
        name: tx.tagName,
        value: tx.amount
    }))
}

export async function createTransaction(data) {
    const response = await api.post('/api/transaction/create', data)
    return response.data
}